.github/workflows/cd.yaml

```
@@ -160,14 +168,32 @@ jobs:
         participant: [company1, company2]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/download-artifact@v3
+        name: 'Download terraform folder'
+        with:
+          name: terraform-${{ matrix.participant }}
+          path: deployment/terraform
 
-      - name: 'Az CLI login'
-        uses: azure/login@v1
+      - name: 'Delete terraform resources'
+        run: |
+          : # Make terraform provider binaries executable because file permissions of uploaded artifacts are not maintained.
+          chmod -R u+x .terraform/providers
```

nit (just for simplicity)
```suggestion
          chmod -R +x .terraform/providers
```